gradient_accumulator/agc.py

```diff
@@ -1,81 +1,88 @@
 import tensorflow as tf
 
-
 # implementation from: https://github.com/sayakpaul/Adaptive-Gradient-Clipping/blob/main/agc.py  # noqa
+SCALAR = tf.constant([], dtype=tf.int32)
+LINEAR = tf.constant([0], dtype=tf.int32)
+TENSOR2D = tf.constant([0, 1], dtype=tf.int32)
+TENSOR3D = tf.constant([0, 1, 2], dtype=tf.int32)
+TENSOR4D = tf.constant([0, 1, 2, 3], dtype=tf.int32)
+
+
+@tf.function
 def compute_norm(x, axis, keepdims):
     """
     Computes the euclidean norm of a tensor :math:`x`.
-
-    Args:
-        x: input tensor.
```

Comment:
I use autodocs to create documentations, so I do not want to remove the `Args:` and `Returns:` statements in the comment here. Same applies to the rest of the methods and classes were this has been done.

Reply:
Noted. I've actually gone ahead and removed the `agc` changes for now to keep this PR better scoped. Can submit a follow-up PR, and will retain the docstring there.